plugins/spamassassin.js

```diff
@@ -186,7 +188,10 @@ exports.do_header_updates = function (connection, spamd_response) {
     }
 
     var modern = plugin.cfg.main.modern_status_syntax;
+    var add_headers = plugin.cfg.main.add_headers;
+
```

Comment:
rather than uselessly iterating through the spamd_response headers, might it be better to return here on line 191 with something like: `if (!plugin.cfg.main.add_headers) return;`?